### ISSUE TITLE
error in gembs index

### ISSUE THREAD

**Issue by sitag** [heathsc/gemBS#65]
It looks like gembs index is  erring out with a gemBS.json not found. Any ideas?

```
Singularity gembs_latest.sif:/output> gemBS index ;  head  mcf10a_ihec.conf ; ls /ref
Traceback (most recent call last):
  File "/usr/local/bin/gemBS", line 11, in <module>
    load_entry_point('gemBS==3.3.2', 'console_scripts', 'gemBS')()
  File "/usr/local/lib/python3.5/dist-packages/gemBS/commands.py", line 157, in gemBS_main
    raise CommandException("gemBS JSON file not found.")
gemBS.utils.CommandException: gemBS JSON file not found.
# Required section
#
# Note that the index and contig_sizes files are generated from the
# reference file if they do not already exist
#
#reference = reference/sacCer3.fa.gz
#reference = /ext/disk1/Homo_sapiens/hg19a/genome/fasta/hg19a.fa
reference = /ref/GCA_000001405.15_GRCh38_no_alt_analysis_set.fasta

#
GCA_000001405.15_GRCh38_no_alt_analysis_set.fasta  conversion_control.fa
```

**Comment by sitag** [comment]
I am also seeing, all other steps seem to work fine (?) 

```
Singularity gembs_latest.sif:/output> gemBS  --loglevel debug prepare -c mcf10a-2.conf -t mcf10a.csv
2019-10-24 01:26:03,747 DEBUG: Using bundled binary : /usr/local/lib/python3.5/dist-packages/gemBS/gemBSbinaries/gemBS_cat
2019-10-24 01:26:03,748 INFO: Starting:
        /usr/local/lib/python3.5/dist-packages/gemBS/gemBSbinaries/gemBS_cat /ref/conversion_control.fa | grep ^>
2019-10-24 01:26:03,748 DEBUG: Starting subprocess
2019-10-24 01:26:03,754 DEBUG: Setting process input to parent output
2019-10-24 01:26:03,754 DEBUG: Starting subprocess
Traceback (most recent call last):
  File "/usr/local/bin/gemBS", line 11, in <module>
    load_entry_point('gemBS==3.3.2', 'console_scripts', 'gemBS')()
  File "/usr/local/lib/python3.5/dist-packages/gemBS/commands.py", line 159, in gemBS_main
    instances[args.command].run(args)
  File "/usr/local/lib/python3.5/dist-packages/gemBS/production.py", line 112, in run
    prepareConfiguration(text_metadata=args.text_metadata,configFile=args.config,no_db=args.no_db,dbfile=args.dbfile,output=args.output)
  File "/usr/local/lib/python3.5/dist-packages/gemBS/__init__.py", line 449, in prepareConfiguration
    line = reader.next()
AttributeError: '_csv.reader' object has no attribute 'next'
Singularity gembs_latest.sif:/output>
Singularity gembs_latest.sif:/output>
Singularity gembs_latest.sif:/output>
Singularity gembs_latest.sif:/output> head -1 mcf10a-2.conf mcf10a.csv
==> mcf10a-2.conf <==
### RUNNING PARAMETERS

==> mcf10a.csv <==
Barcode,Name,Library,File1,File2
```

**Comment by sitag** [comment]
It seems like a python 2 vs 3 issue: https://stackoverflow.com/questions/42767250/python-csv2libsvm-py-attributeerror-csv-reader-object-has-no-attribute-nex ; I am running singularity with --clean-env , so I don't think my environment is leaking into there.

**Comment by sitag** [comment]
@heathsc It seem like https://github.com/heathsc/gemBS/blob/d1f917426bcc1e804cf808edb4d51782e9eaff46/gemBS/__init__.py#L435-L451 reader.next() is a bug. I can avoid that branch of if by adding in one more field to my csv and then everything works.

**Comment by heathsc** [comment]
This is a legacy pathway for the original config file for gemBS v1 that I think was only ever used internally and is not documented.  This config file had no header line and a fixed 5 column format.  Normally this pathway should only be taken if a valid header line is not found.  I have removed this branch so that if a valid header is not found the user will get an error about an invalid config file, rather than gemBS trying to interpret the file as a legacy config file which risks leading to confusing error messages for the user.  

In short, the python2 usage of reader.next() is a clear bug but was not picked up in testing as this pathway was not triggered...  This issues should be fixed in 3.3.3
